### DISCUSSION TITLE
Do not open the script when opening the scene

### DISCUSSION THREAD

**Discussion by ditiem-games** [godotengine/godot-proposals#10446] [GUI]
When you double click on a scene, the root node script is opened in the Script tab, even if the Script tab is not the selected one!

I know some use case can come out with this behavior but I feel rare (read furious) that the system does something I did not ask for. If I want to open the script I would click on the script file or I would do another click on the root-node-script icon, otherwise the small selected set of scripts I have opened keep growing.

Another very strong reason I found is that when there are many scenes and / or scripts opened, launching the scene to test takes ages.

**Comment by Calinou** [comment]
You can uncheck **Open Dominant Script on Scene Change** in the Editor Settings to disable this behavior.

> Another very strong reason I found is that when there are many scenes and / or scripts opened, launching the scene to test takes ages.

You can disable **Save Before Running** to make running the project faster when many scenes are open, although you'll need to remember to save your changes before running.

**Reply by ditiem-games** [reply]
Awesome! UX has increased 20% :). Thank you a lot for the tip!

I am not sure the "save before running" is the issue, as it is not only the launch time, it is also the time when you switch back to godot editor to see the remote information (I am talking about 1 min waiting till it responds). I will test it anyway.

**Reply by Calinou** [reply]
Note that **Open Dominant Script on Scene Change** is disabled by default since 4.3: https://github.com/godotengine/godot/pull/81906

**Reply by ditiem-games** [reply]
> You can uncheck **Open Dominant Script on Scene Change** in the Editor Settings to disable this behavior.

Would not make sense to remove the "tabs" when you are in script mode if this option of **Open Dominant Script on Scene Change** is disabled? It is just so confusing having tabs there that do nothing.

**Reply by Calinou** [reply]
Scene tabs still have an importance when in the script editor. The currently selected scene tab determines the context used for scene-specific autocompletion, such as `get_node()` and `$` (or any other method that expects Node or NodePath parameters).

**Reply by ditiem-games** [reply]
If I understood you well, that makes no sense to me. `get_node()` and `$` should autocomplete based on the current select scene and the position of the node in which the script is attached to. If I am in another scene where that script has nothing to do there, I dont see any autocompletion there. In fact, it should keep the original autocompletion context if the script is not attached to more than one node and it is decidable.

You do not need the tabs when you are in the *script tab*, specially if they do not autoselect the main script file when changing the scene. 

My idea is that they will be visible when you are in the 2D and 3D tab, and in the script only if the *Open Dominant Script on Scene Change* is checked.

All those scene tabs could be nicely replaced by a breadcrumb with memory on top of the screen dock area, that will fix all the problems in one shot, but that is another discussion.
